cli/version/version.go

```diff
@@ -41,16 +44,133 @@ func BuildVersionString() string {
+
+	if showRocksVersion {
+		if err := printRocksVersion(projectPath); err != nil {
+			return fmt.Errorf("%s: %s", rocksVersionsGetError, err)
+		}
+	}
+
+	return nil
+}
+
+func formatVersion(template string, templateArgs map[string]string) string {
+	versionMsg, err := templates.GetTemplatedStr(&template, templateArgs)
+
+	if err != nil {
+		panic(err)
+	}
+
+	return versionMsg
+}
+
+func getRocksVersions(projectPath string) (map[string]string, error) {
+	var rocksVersionsMap map[string]string
+	var err error
+
+	if fileInfo, err := os.Stat(projectPath); os.IsNotExist(err) {
+		return nil, fmt.Errorf("Specified project path doesn't exist")
+	} else if err != nil {
+		return nil, fmt.Errorf("Impossible to use specified project path: %s", err)
+	} else if !fileInfo.IsDir() {
+		return nil, fmt.Errorf("Specified project path %s is not a directory", projectPath)
+	}
+
+	if rocksVersionsMap, err = common.LuaGetRocksVersions(projectPath); err != nil {
+		return nil, err
+	}
+
+	return rocksVersionsMap, nil
+}
+
+func printCartridgeVersion(projectPath string) error {
+	var rocksVersions map[string]string
+	var err error
+
+	if rocksVersions, err = getRocksVersions(projectPath); err != nil {
+		return err
+	}
+
+	if rockspecPath, err := common.FindRockspec(projectPath); err != nil {
+		return err
+	} else if rockspecPath == "" {
+		return fmt.Errorf("Project path %s is not a project", projectPath)
+	}
+
+	version := rocksVersions["cartridge"]
+	if version == "" {
+		return fmt.Errorf("Are dependencies in .rocks directory correct?")
+	}
+
+	cartridgeVersion := formatVersion(cartridgeVersionTmpl, map[string]string{
+		"Title":   cartridgeVersionTitle,
+		"Version": version,
+	})
+
+	fmt.Println(cartridgeVersion)
+	return err
+}
+
+func printRocksVersion(projectPath string) error {
+	var versionParts []string
+	var rocksVersions map[string]string
+	var err error
+
+	if rocksVersions, err = getRocksVersions(projectPath); err != nil {
+		return err
+	}
+
+	if len(rocksVersions) == 0 {
+		return fmt.Errorf(`Looks like your project directory
+does not contain a .rocks directory... Did you built your project?`)
+	}
 
-	osArchStr := fmt.Sprintf("%s/%s", runtime.GOOS, runtime.GOARCH)
-	versionParts = append(versionParts, osArchStr)
+	versionParts = append(versionParts, "\nRocks")
 
-	if gitCommit != "" {
-		gitCommitStr := fmt.Sprintf("commit: %s", gitCommit)
-		versionParts = append(versionParts, gitCommitStr)
+	for rock, version := range rocksVersions {
+		// We have to skip cartridge rock - we print info about
+		// this rock in function above.
+		if rock != "cartridge" {
+			versionParts = append(versionParts, fmt.Sprintf("%s %s", rock, version))
+		}
 	}
 
-	return strings.Join(versionParts, " ")
+	fmt.Println(strings.Join(versionParts, "\n "))
+	return nil
 }
+
+var (
+	cliVersionTmpl = `{{ .Title }}
+ Version:	{{ .Version }}
+ OS/Arch: 	{{ .OS }}/{{ .Arch }}
+ Git commit:	{{ .Commit }}
+`
+
+	cartridgeVersionTmpl = `{{ .Title }}
+ Version:	{{ .Version }}`
```

Comment:
Let's add here `\n` and remove it from `versionParts = append(versionParts, "\nRocks")`.

Reply:
We will get a empty line (after all prints) in case if --rocks flag not specified, but, if ``--rocks`` flag specified, we don't get empty line. Is it okay?